docker/lib/dependabot/docker/tag.rb

```diff
@@ -51,47 +60,55 @@ def comparable_to?(other)
         equal_prefix && equal_format && equal_suffix
       end
 
+      sig { returns(T::Boolean) }
       def comparable?
         name.match?(NAME_WITH_VERSION)
       end
 
+      sig { params(other: Tag).returns(T::Boolean) }
       def same_precision?(other)
         other.precision == precision
       end
 
+      sig { params(other: Tag).returns(T::Boolean) }
       def same_but_less_precise?(other)
         other.segments.zip(segments).all? do |segment, other_segment|
           segment == other_segment || other_segment.nil?
         end
       end
 
+      sig { returns(T::Boolean) }
       def canonical?
         return false unless numeric_version
         return true if name == numeric_version
 
         # .NET tags are suffixed with -sdk
-        return true if name == numeric_version + "-sdk"
+        return true if name == T.must(numeric_version) + "-sdk"
 
-        name == "jdk-" + numeric_version
+        name == "jdk-" + T.must(numeric_version)
```

Comment:
Similar to above
```suggestion
        numeric_version && name == "jdk-" + numeric_version
```

If not, keep it as is.